docs/get-started/cli.md

```diff
@@ -4,4 +4,219 @@ title: On your local machine
+
+## Test your Pack locally
+
+One of the advantages of developing locally is that you can test your Pack code without having to upload it to Coda's servers. Let's test the new `Hello` formula you just added:
+
+```shell
+npx coda execute manifest.ts Hello "world"  
+```
+
+If everything works correctly this should output `Hello world!`.
+
+## Upload your Pack
+
+So far everything you've built only exists on your local machine, and Coda has no knowledge of it. To see it working in a real doc you'll need to upload your Pack to Coda's servers.
+
+### Register an API token
+
+The `coda` CLI uses the Coda API under the hood to upload your code, and likewise needs an API token to operate. Registering an API token is a one-time setup step that must be done for each new Pack.
+
+1. Register an API key for Pack uploads:
+
+    ```shell
+    npx coda register --codaApiEndpoint=https://{{coda.domain}}
+    ```
+
+1. When prompted to create a new API token, type `y` and hit enter.
+
+    This will open your browser to the API token creation dialog.
+
+1. In the **Name** field enter "Hello World Pack" and then click **Generate API token**.
+
+1. Scroll down the page to the **Coda API tokens** section, find the token you just created, and click the **Copy token** link.
+
+1. Switch back to your terminal, paste your token into the prompt, and hit enter.
+
+This will create a new file `.coda.json` in your working directory that contains the API token.
+
+??? warning "Don't check in .coda.json"
+    If you use a version control system you will likely want to make sure this file isn't check in, as the token within provides access to your account. For example, if using Git, add `.coda.json` to your [`.gitignore` file](https://git-scm.com/docs/gitignore).
+
+### Create the Pack
+
+Now that you have the access configured you can create the new Pack on Coda's servers. Like registering an API token, this is also a one-time setup step that needs to be done for each Pack.
+
+```shell
+npx coda create manifest.ts --name "Hello World" --description "My first Pack." --codaApiEndpoint=https://{{coda.domain}}
+```
+
+This will create a new, empty Pack on Coda's servers and output its URL in the Pack Studio. It stores the Pack's ID in the new file `.coda-pack.json`.
+
+### Upload your first version
+
+Now that you've established access and created the empty Pack, you're finally ready to upload your code.
+
+```shell
+npx coda upload manifest.ts --notes "Initial version." --codaApiEndpoint=https://{{coda.domain}}
+```
+
+??? note "Source code not avilable"
+    If you open your Pack in the online Pack Studio code editor you'll see a message like:
+    
+    ```ts
+    // Some boilerplate when failed to load source code
+    ```
+
+    This is expected, since the CLI will only upload the built Pack and not the source code.
+
+## Install and use the Pack
+
+Your new Pack is now available to use in all your docs, and you can install it just like a Coda-made Pack.
```

Comment:
Nit, I wonder if "Coda-made Pack" will cease to mean as much going forward when most packs are user-created.

Reply:
Good point. Fixed here and in `web.md`